src/hotspot/share/classfile/systemDictionary.cpp

```diff
@@ -1612,29 +1607,32 @@ InstanceKlass* SystemDictionary::find_or_define_instance_class(Symbol* class_nam
 
 // ----------------------------------------------------------------------------
 // Update hierarchy. This is done before the new klass has been added to the SystemDictionary. The Compile_lock
-// is held, to ensure that the compiler is not using the class hierarchy, and that deoptimization will kick in
-// before a new class is used.
+// is held, to ensure that the compiler is not using the class hierarchy.
```

Comment:
nit: s/is held/is grabbed/
Adjust the header comment to reflect that we now grab the Compile_lock
in the function rather than expecting it to be grabbed by the caller.

Reply:
Fixed, thanks!